cmplxs/cscalar/cscalar.go

```diff
@@ -0,0 +1,91 @@
+// Copyright ©2013 The Gonum Authors. All rights reserved.
+// Use of this code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cscalar
+
+import (
+	"math"
+	"math/cmplx"
+
+	"gonum.org/v1/gonum/floats/scalar"
+)
+
+// EqualWithinAbs returns true if a and b have an absolute difference
+// not greater than tol.
+func EqualWithinAbs(a, b complex128, tol float64) bool {
+	return a == b || cmplx.Abs(a-b) <= tol
+}
+
+const minNormalFloat64 = 2.2250738585072014e-308
+
+// EqualWithinRel returns true if the difference between a and b
+// is not greater than tol times the greater absolute value of a and b,
+//  abs(a-b) <= tol * max(abs(a), abs(b)).
+func EqualWithinRel(a, b complex128, tol float64) bool {
+	if a == b {
+		return true
+	}
+
+	delta := cmplx.Abs(a - b)
+	if delta <= minNormalFloat64 {
+		return delta <= tol*minNormalFloat64
+	}
+	// We depend on the division in this relationship to identify
+	// infinities.
+	return delta/math.Max(cmplx.Abs(a), cmplx.Abs(b)) <= tol
+}
+
+// EqualWithinAbsOrRel returns true if parts of a and b are equal to within
+// the absolute or relative tolerances.
+func EqualWithinAbsOrRel(a, b complex128, absTol, relTol float64) bool {
+	return EqualWithinAbs(a, b, absTol) || EqualWithinRel(a, b, relTol)
+}
+
+// ParseWithNA converts the string s to a complex128 in v.
+// If s equals missing, w is returned as 0, otherwise 1.
```

Comment:
This is quite cryptic. Could the doc explain what is the purpose of `w`?

Reply:
It indicates a weight, but I don't think it's cryptic, it says what it does.

Reply:
Can it say that `w` is a weight? Knowing that, everything makes sense to me now.

Reply:
Please read it in the context of a godoc rendering (the code it old, so you can see it [here](https://godoc.org/gonum.org/v1/gonum/floats#ParseWithNA)). Does it still need to be expanded?

Part of the reason I'm pushing back is that there's a philosophical (in the correct sense) issue with saying that `w` is a weight; it may be used as one, and that is the intention, but it's not a weight, it is a conditional value that is `float64(0)` when the input matches the missing value text and `float64(1)` otherwise. It *can* be used as a weight, and this is demonstrated in the example immediately below the godoc signature and description.

Reply:
But when I'm reading the code (say, when refactoring this file), I don't see the godoc rendering. IMHO locality of information is useful for code readability.

Why not just add what you wrote above: "w is intended to be used as a weight for the parsed value"?

Reply:
Expanded reluctantly.

Reply:
Thanks.